src/test/java/io/supertokens/test/userMetadata/UserMetadataTest.java

```diff
@@ -0,0 +1,173 @@
+import io.supertokens.usermetadata.UserMetadata;
+
+import org.junit.*;
+import org.junit.rules.TestRule;
+
+import static io.supertokens.test.passwordless.PasswordlessUtility.*;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import com.google.gson.JsonNull;
+import com.google.gson.JsonObject;
+
+/**
+ * This UT encompasses tests related to update user
+ */
+public class UserMetadataTest {
+
+    @Rule
+    public TestRule watchman = Utils.getOnFailure();
+
+    @AfterClass
+    public static void afterTesting() {
+        Utils.afterTesting();
+    }
+
+    @Before
+    public void beforeEach() {
+        Utils.reset();
+    }
+
+    /**
+     * gets the metadata for a user who doesn't have one
+     *
+     * @throws Exception
+     */
+    @Test
+    public void getEmptyMetadata() throws Exception {
+        String[] args = { "../" };
+
+        TestingProcessManager.TestingProcess process = TestingProcessManager.start(args);
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STARTED));
+
+        if (StorageLayer.getStorage(process.getProcess()).getType() != STORAGE_TYPE.SQL) {
+            return;
+        }
+
+        String userId = "userId";
+        JsonObject metadata = UserMetadata.getUserMetadata(process.getProcess(), userId);
+
+        assertNotNull(metadata);
+        assertEquals(metadata.entrySet().size(), 0);
+
+        process.kill();
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STOPPED));
+    }
+
+    /**
+     * set the metadata for a user who doesn't have one
+     *
+     * @throws Exception
+     */
+    @Test
+    public void createMetadata() throws Exception {
+        String[] args = { "../" };
+
+        TestingProcessManager.TestingProcess process = TestingProcessManager.start(args);
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STARTED));
+
+        if (StorageLayer.getStorage(process.getProcess()).getType() != STORAGE_TYPE.SQL) {
+            return;
+        }
+
+        String userId = "userId";
+        JsonObject update = new JsonObject();
+        update.addProperty("test", "123");
+        update.add("testNull", JsonNull.INSTANCE);
+        JsonObject updateResult = UserMetadata.updateUserMetadata(process.getProcess(), userId, update);
+
+        JsonObject newMetadata = UserMetadata.getUserMetadata(process.getProcess(), userId);
+
+        assertEquals(updateResult, newMetadata);
+        assertEquals(1, updateResult.entrySet().size());
+        assertEquals("123", updateResult.get("test").getAsString());
+        assert (!updateResult.has("testNull"));
+
+        process.kill();
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STOPPED));
+    }
+
+    /**
+     * updates the metadata of a user and merges the result
+     *
+     * @throws Exception
+     */
+    @Test
+    public void updateMetadata() throws Exception {
```

Comment:
You should also add a test to test concurrent updates to a user's metadata to ensure race conditions are handled correctly.